python/lsst/daf/butler/registry/queries/_builder.py

```diff
@@ -253,7 +253,18 @@ def joinDataset(self, datasetType: DatasetType, collections: Any, *,
                 self._doomed_by.extend(rejections)
             else:
                 self._doomed_by.append(f"No collections to search matching expression {collections}.")
-            return False
+            # Make a single subquery with no collections that never yields
+            # results; this should never get executed, but downstream code
+            # still needs to access the SQLAlchemy column objects.
+            ssq = datasetRecordStorage.select(
+                dataId=SimpleQuery.Select,
+                id=SimpleQuery.Select if isResult else None,
+                run=SimpleQuery.Select if isResult else None,
+                ingestDate=SimpleQuery.Select if isResult else None,
+            )
+            if findFirst:
+                ssq.columns.append(sqlalchemy.sql.literal(rank).label("rank"))
```

Comment:
I guess we don't expect any dimension to be named `rank`?

Reply:
That is indeed a limitation, but it's not a new one on this branch (just shuffled code around).